lib/barrier.c

```diff
@@ -45,7 +45,20 @@ void barrier_spin_wait(barrier_t *barrier)
     if (__sync_sub_and_fetch(&barrier->count, 1) != 0) {
         volatile bool *i_am_blocked = &waiting_flags[my_cpu].flag;
         while (*i_am_blocked) {
+#if defined(__x86_64) || defined(__i386__)
             __builtin_ia32_pause();
+#elif defined (__loongarch_lp64)
+            __asm__ __volatile__ (
```

Comment:
If/when some pause-type instruction is added to a future implementation of the architecture, the code can be adjusted.

Reply:
Yes, if/when the pause instruction has been added, I will adjust this code.